### ISSUE TITLE
Adding first-seen/last-seen to dataset with new flags

### ISSUE THREAD

**Issue by codingo** [codingo/dooked#1]
Whilst the captured data is good, it fails in conditions where there's load balancing in place (as  the IP will rotate on each capture).

Suggested fix to this is to add two date fields, `first-seen` and `last-seen`, potentially also `seen` as an integer count (less valuable, nice to have).

As alerts would then change around this, flags should be added:

- `fs` which would boolean toggle a messaging being shown each run of all first scene items
- `ls` which would toggle messaging to show if an item was in the last scan, but is no longer seen in days since the specified run. So  `ls 2` woudl shown any items not seen this run, or in the last two days.
- `lsd` which is ls, but in US datetime format, showing prompts for items that haven't been seen since `lsd <date>`

Bounty for this item is $100USD. If you put in a pull request, be sure to mention a paypal link (here or in dm on twitter.com/codingo_).

**Comment by mahmudsudo** [comment]
hi , is this actively been worked on ? , can i take on it ?

**Comment by Asobu01** [comment]
I can take codingo/dooked#1.
Start: reproducing the reported issue first, then applying a focused fix with clear validation.
ETA: draft PR in ~12h.
Deliverables: repro + root-cause summary; focused patch (plus tests/docs if needed); review-ready PR with validation notes. @Asobu01

**Comment by Asobu01** [comment]
I can take codingo/dooked#1.
Start: reproducing the reported issue first, then applying a focused fix with clear validation.
ETA: draft PR in ~12h.
Deliverables: repro + root-cause summary; focused patch (plus tests/docs if needed); review-ready PR with validation notes. @Asobu01

**Comment by Asobu01** [comment]
I can take codingo/dooked#1.
Start: reproducing the reported issue first, then applying a focused fix with clear validation.
ETA: draft PR in ~12h.
Deliverables: repro + root-cause summary; focused patch (plus tests/docs if needed); review-ready PR with validation notes. @Asobu01

**Comment by Asobu01** [comment]
I can take codingo/dooked#1.
Start: reproducing the reported issue first, then applying a focused fix with clear validation.
ETA: draft PR in ~12h.
Deliverables: repro + root-cause summary; focused patch (plus tests/docs if needed); review-ready PR with validation notes. @Asobu01

**Comment by Asobu01** [comment]
I can take codingo/dooked#1.
Start: reproducing the reported issue first, then applying a focused fix with clear validation.
ETA: draft PR in ~12h.
Deliverables: repro + root-cause summary; focused patch (plus tests/docs if needed); review-ready PR with validation notes. @Asobu01

**Comment by Asobu01** [comment]
Quick follow-up on `codingo/dooked#1`. If this issue is still open, I can start right away and post a short execution checklist first. (@Asobu01)

**Comment by nexiumbiz-debug** [comment]
PR submitted: https://github.com/codingo/dooked/pull/4

Implemented the first-seen / last-seen history tracking and the requested `--fs`, `--ls`, and `--lsd` comparison flags. I have not posted payout details publicly; if the PR is selected, I can provide them privately through your preferred trusted channel.

**Comment by Lukas616** [comment]
PR submitted here: https://github.com/codingo/dooked/pull/5

I added the requested record history fields and reporting flags, documented the options, and included validation notes in the PR comment.

Payment can be sent here if accepted/merged: https://paypal.me/Lukas61616

**Comment by JeremyZeng77** [comment]
Submitted PR #11 for this bounty: https://github.com/codingo/dooked/pull/11

It adds first-seen / last-seen / seen history fields, preserves missing records in subsequent JSON output, and adds the requested `--fs`, `--ls`, and `--lsd` reporting flags.

If accepted for the $100 bounty, PayPal payout can go to: https://paypal.me/Jeremytsangchina

**Comment by Da7-Tech** [comment]
PR submitted: https://github.com/codingo/dooked/pull/16\n\nThis adds first-seen / last-seen / seen history fields, preserves missing historical records in JSON output, and implements the requested --fs, --ls, and --lsd reporting flags.\n\nIf accepted for the  bounty, PayPal payout can go to: aa.631@hotmail.com

**Comment by jdjioe5-cpu** [comment]
Hi @codingo, I’m interested in this $100 PayPal bounty, but I noticed there are already several PRs linked here.

Before I start a fresh implementation, could you confirm whether new submissions are still being considered for this bounty, and whether PayPal payout is still available if a new PR is accepted and merged?

If yes, I’ll keep the PR focused, include local validation notes, and provide PayPal details only after there is a realistic merge/reward path.

**Comment by jdjioe5-cpu** [comment]
PR submitted: https://github.com/codingo/dooked/pull/19

This implements the requested first-seen / last-seen history fields, preserves missing historical records in JSON output, and adds the requested `--fs`, `--ls`, and `--lsd` flags.

PayPal payout details can be provided privately by DM if this PR is selected and merged.

**Comment by clynbmilio** [comment]
PR submitted: https://github.com/codingo/dooked/pull/20

This implements first-seen / last-seen / seen history, preserves missing historical DNS records in JSON output, and adds the requested `--fs`, `--ls`, and `--lsd` reporting flags.

I also fixed two small current-build compatibility issues found during validation. Evidence: fork GitHub Actions build check passes here: https://github.com/clynbmilio/dooked/actions/runs/27055020294

If selected and merged for the $100 bounty, PayPal payout can go to: versytwo@icloud.com

**Comment by jdjioe5-cpu** [comment]
## Bounty Review Request — PR #19

PR #19 (https://github.com/codingo/dooked/pull/19) adds the `first-seen`/`last-seen` history fields and `--fs`/`--ls`/`--lsd` flags exactly as described in issue #1. The PR is open, MERGEABLE, and ready.

**PR state at 2026-06-18:**
- State: OPEN, mergeable: MERGEABLE
- Head: `63b3dddf` (+280/-10, 4 files)
- Scope: DNS history metadata + 3 CLI flags + README docs + 2 build-compat fixes
- Diff: `git diff origin/main...HEAD` is clean

**Payout readiness:**
The PR body states: *"PayPal payout details can be provided privately by DM if this PR is selected and merged."* I can provide PayPal details via DM immediately upon merge.

**Verification:**
`cmake . && make -j2 && ./dooked --help` should show the new `--fs`/`--ls`/`--lsd` flags.

Bounty #19, `codingo/dooked`.
